### PR TITLE
[READY (again)]Adds a new makeshift weapon: the pipe pistol

### PR DESCRIPTION
**Pull request by Tacolizard**
**Should be good to go now. The DNM tag was added by the bot and won't go away.**

:cl: Tacolizard Forever
add: Added a new makeshift pistol, the pipe pistol.
add: Added a new makeshift ammo type, the makeshift 9mm round.
/:cl:

- Fully sprited
- Is compatible with 9mm rounds
- uses an internal mag with 9 rounds
- makeshift 9mm rounds do 7 damage
- ~~Has three different sprites that are chosen randomly when the gun is created~~
- Uses WJohn sprites

**no longer uses variant sprites, only uses one sprite made by WJohn**

I have also added a spark effect that is activated by the `sparks` var. When set to `1`, a gun will shoot sparks every time it fires. The pipe pistol has `sparks = 1`.

This gun was made to be a low-damage alternative the improvised shotgun, and its sprite is themed similarly. With 9 rounds that do 7 damage, it is powerful but can't crit, knockdown or kill. This gun costs as much to build as an improvised shotgun, and the ammo is crafted the same as the improvised shotgun shells. This gun is designed to offer an alternative gameplay style to the people who already craft weapons.

If there are balance concerns, please mention them. I may be willing to compromise.

(mag size was 6, but was raised to 9 because the 7-damage rounds are very weak)

For comparison, a single buckshot pellet does 12.5 damage. iirc a stetchkin round does about 20 damage.

**Uses the sprite on the left**
pic

**hmm look at the upvotes and downvotes. Coders hate this but players and admins seem to like it. What say you, players? You like having the coderbus dictatorship influence your game????????**

**also kevinz please whitelist my prs on your auto-downvote bot :^)**

### CODE REVIEW

**Comment by anconfuzedrock** [comment]
can't cargo already buy guns with riot foam darts or buy the guns and fill them with pens for 5 brute?

**Comment by Tacolizard** [comment]
@anconfuzedrock iirc that requires an emag to accomplish. Also, this gun isn't meant to compete with traitor items, or be used by traitors.

**Comment by Repukan** [comment]
Make the ammo makeshift too

**Comment by Repukan** [comment]
Nevermind I misread

**Comment by TechnoAlchemisto** [comment]
This is just a 9 round stechkin for assistants

**Comment by XDTM** [comment]
Definitely not overpowered, 7 damage is about 35% of a normal stechkin damage which is not really that high of its own, and it costs pretty much as much as a shotgun. I think it's a nice ghetto weapon.

**Comment by TechnoAlchemisto** [comment]
most people will not fall for the pitfall that is improvised ammo, this is compatible with real bullets.

**Comment by sawrge** [comment]
Why?

**Comment by sawrge** [comment]
Like seriously why add more fucking easy access ranged weapons? What purpose does this serve? What does this bring to the game?

**Comment by Repukan** [comment]
It adds a makeshift gun

**Comment by dannno** [comment]
goofy
probably unnecessary

**Comment by Tacolizard** [comment]
@sawrge 

> This gun was made to be a low-damage alternative the improvised shotgun, and its sprite is themed similarly. With 9 rounds that do 7 damage, it is powerful but can't crit, knockdown or kill.

it fits a niche that the i-shotgun doesn't fill. It's a nonlethal semi-auto to be used by the crew against ops, or by cult in the early game. The i-shotgun fits the high damage makeshift niche.

**Comment by Tacolizard** [comment]
@TechnoAlchemisto this is compatible with regular 9mm bullets that are only obtainable from traitors or a hacked autolathe. Even with 9mm bullets loaded, this gun is inferior to the stetchkin due to its lower round capacity and inability to load external mags. Needing to fiddly load individual rounds is part of the balance here.

**Comment by Tacolizard** [comment]
This gun is compatible with regular 9mm just as the improvised shotgun is compatible with buckshot.

**Comment by ExcessiveUseOfCobblestone** [comment]
make it look like a welrod and you'll have me sold :^)

**Comment by Shapsy** [comment]
The difference is you can't load a full magazine into an improvised shotgun, this is just a stechkin for assistants because hacked lathes are piss easy to come by

**Comment by ExcessiveUseOfCobblestone** [comment]
but you can't load a full mag, you have to individually add the bullets into it.

**Comment by Tacolizard** [comment]
then mabye sec should do their job and keep people from hacking lathes, which are coincidentally also a source of buckshot for the improvised shotgun

**Comment by octareenroon91** [comment]
Possible balance concerns:
No risk of failure; doesn't cost wood.

**Comment by Tacolizard** [comment]
@octareenroon91 I was just thinking that. I'll make it require some wood planks. However I don't know what you mean by 'risk of failure'. the improvised shotgun, what i assume you are comparing it to, doesn't use rng to determine if a shot fires.

**Comment by Tacolizard** [comment]
> make it look like a welrod and you'll have me sold :^)

I might do that actually

**Comment by ma44** [comment]
If a cargo tech has a emag, this becomes a free 9 round stechkin

Also ammo isn't a concern if you can just build the box, refill bullets then slap it back into the autolathe

**Comment by Tacolizard** [comment]
@ma44 if they plan on murdering someone, the better option by far is to buy a real stetchkin. A real stetchkin can use mags, can be obtained quicker and easier, and has a higher capacity. Also I believe the stetchkin uses 10mm, not 9mm. 

You could make the same argument against the improvised shotgun, ma44.

**Comment by MMMiracles** [comment]
Add inaccuracy spread so it isn't just a ghetto Sketchin.

**Comment by Tacolizard** [comment]
heres my crappy welrod attempt. I just made it for fun.

welrod

**Comment by Tacolizard** [comment]
@MMMiracles 

How would I do that? Do I need to add a var to the gun or the ammo?

**Comment by MMMiracles** [comment]
There's an inaccuracy variable for ammo that forces random spread.

**Comment by Tacolizard** [comment]
if its on ammo then regular 9mm wouldn't have spread, only the makeshift ones would which kinda defeats the purpose.

**Comment by MMMiracles** [comment]
Then have the weapon have a chance to backfire on the user if they use the non-makeshift ammo while keeping inaccuracy on the makeshift rounds.

**Comment by FantasticFwoosh** [comment]
Shiver my timbers, looks like a flintlock! Space pirates should get a variant of this for sure.

**Comment by Tacolizard** [comment]
@MMMiracles I do like that idea, but I'm hesitant to use RNG on anything because people seem to hate it.

**Comment by Armhulen** [comment]
Don't change anything, it's great as is.

**Comment by Fox-McCloud** [comment]
Given cargo can already purchase riot stetchkins (no emag required) and get tons of riot darts, well...this doesn't seem that bad, to be honest.

**Comment by Supermichael777** [comment]
ill be sure to consider this next time i want to TICKLE someone to death

File: code/modules/crafting/recipes.dm
Comment (Fox-McCloud) [inline code/modules/crafting/recipes.dm]: I'd drop the chem grenade requirement; for the shotgun shell, that makes some sense, but this is a tiny round.

just having the metal+cable coil+fuel requirement is going to make this annoying enough.

**Comment by Bawhoppen** [comment]
isn't the problem that one person can carry like 9 of these and spam spray them even if they're low damage?

**Comment by Fox-McCloud** [comment]
I mean, technically yes, but in a day and age where the lowest rank on station can make a two-shot air cannon, I don't know how much relevance that has.

**Comment by Bawhoppen** [comment]
don't take my critique as dismissal, I like this

also you should make it randomly choose between the two sprites on crafting it

File: code/modules/crafting/recipes.dm
Comment (Tacolizard) [inline code/modules/crafting/recipes.dm]: i do want to drop the chem grenade, but if i understand correctly the chem grenade is what holds the fuel. So if i remove the chem grenade i also have to remove fuel from it.

**Comment by Tacolizard** [comment]
@Bawhoppen that's a good idea, but exactly how crappy do people think the welrod looks? If people like the welrod sprite then yeah i'll do it.

**Comment by Bawhoppen** [comment]
welrod is fine, make another variant too

**Comment by Tacolizard** [comment]
alright, I'm on it

**Comment by Tacolizard** [comment]
lol i forgot a paranthesis

**Comment by Bawhoppen** [comment]
another idea: make it so you can take a prescription glasses lens out and use it as a scope

File: code/modules/crafting/recipes.dm
Comment (octareenroon91) [inline code/modules/crafting/recipes.dm]: I'm pretty sure tablecraft just removes chems as available.

**Comment by octareenroon91** [comment]
> another idea: make it so you can take a prescription glasses lens out and use it as a scope

This sounds pretty strong to restrict to the sketchy-in.

**Review by Bawhoppen — APPROVED** [review]

File: code/modules/crafting/recipes.dm
Comment (XDTM) [inline code/modules/crafting/recipes.dm]: You just need to have the chem in a beaker near you, and it'll get used in the recipe, don't worry.

**Comment by XDTM** [comment]
As a balance mechanism, the gun could take damage when shooting bullets that aren't made for it (i.e. regular ones), breaking it down after about a full mag of shooting. That way it's not RNG, and you have the option to shoot hard but lose the gun for good.

**Comment by Tacolizard** [comment]
@XDTM I would do that, but it seems most people are satisfied with it in its current form.

**Comment by Tacolizard** [comment]
So everything works, except that the gun's description never updates when fired. it just always says there are 9 rounds remaining

**Comment by Tacolizard** [comment]
ok everything is fixed. Ready for merge

**Comment by Tacolizard** [comment]
someone should probably make that bot remove the Do Not Merge tag when [DNM] is taken out of the title

**Comment by Bawhoppen** [comment]
This is the 3rd sprite variant made by me: 
ss 2017-04-26 at 02 08 28

**Comment by Armhulen** [comment]
Have it randomize between the three

**Comment by Tacolizard** [comment]
it does

**Comment by Niknakflak** [comment]
I like it

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (octareenroon91) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: The spelling is "breech" not "breach". Please fix if you add more commits.

**Comment by Ausops** [comment]
If this is a makeshift weapon that is hard to reload, it should consider players making and carrying several of these. Consider bags full of 9shot weapons firing 9mm.

**Comment by Bawhoppen** [comment]
@Ausops I already mentioned it, and the solution was he added an RNG chance to backfire
At least I think he did

**Comment by WJohn** [comment]
There are so many things wrong with the sprites I don't even know where to start.

Also balance memes is probably gonna block this forever.

**Comment by Ausops** [comment]
same here :^3c

**Comment by lzimann** [comment]
I particularly don't like adding makeshift guns like this easily accessible, might aswell make everyone spawn with a gun...
Anyway, @KorPhaeron probably your call on this.

**Comment by Tacolizard** [comment]
what happened with travis? this compiled locally fine

**Comment by Tacolizard** [comment]
@WJohn I thought the welrod was crappy, I only added it because people seemed to like it somehow. I'll use your sprites.

**Comment by Tacolizard** [comment]
@Ausops @Bawhoppen No, I did not add any RNG. I will make this only fit on the belt or something like that to combat bags of em. I don't like rng

**Comment by MMMiracles** [comment]
the best way to get good sprites is to bait out the spriters with codersprites.

**Comment by Tacolizard** [comment]
@WJohn I like your sprites, but this gun doesn't use magazines. It is fed bullets individually.

**Comment by WJohn** [comment]
You really should have updated your PR description... I was so confused on what it was supposed to do given how many times it's been changed.

In that case, just use the first sprite without the magazine.

**Comment by Tacolizard** [comment]
@WJohn The PR description never says it uses magazines

**Comment by GunHog** [comment]
Those Ausops sprites...

**Comment by Tacolizard** [comment]
@lzimann my logic is that if the improvised shotgun is fine, a low power pistol for roughly the same cost is fine as well.

**Comment by lzimann** [comment]
But I don't think the improvised shotgun is fine(not like it was ever used )

**Comment by Tacolizard** [comment]
I see it used sometimes. It's most often used by revs in my experience

**Comment by kevinz000** [comment]
first of all, it's not coderbus dictatorship
second of all, i don't like the idea of everyone ever having an improvised weapon
enough fucking greytide without assistants having makeshift pistols to shoot at me when i'm trying to arrest them and they have the advantage because i'm chasing them.

**Comment by KorPhaeron** [comment]
Is it petty that I was thinking about merging this but am probably now just going to ignore it until it dies because of the REBEL AGAINST THE CODERS thing in the OP

**Comment by Bawhoppen** [comment]
yeah, he probably shouldn't have done that

it's not ready 2 be merged anyways

**Comment by ExcessiveUseOfCobblestone** [comment]
THE WELROD WAS FUNNY!

**Comment by PKPenguin321** [comment]
it'll be fiiine just merge it

**Comment by lzimann** [comment]
@tgstation/commit-access

**Comment by optimumtact** [comment]
R E B E L

**Comment by Tacolizard** [comment]
coderbussistan wins again. You have won the battle, but you won't win the war

**Comment by Tacolizard** [comment]
the one time i make a popular non-fix pr, it gets closed as a meme.

**Comment by Bawhoppen** [comment]
the people who have to merge it are coders you realise

**Comment by optimumtact** [comment]
this didn't generate the amount of funny salt I thought it would. sad

**Comment by lzimann** [comment]
I am still closing this if no other maintainer express their interest/merge this.

**Comment by Bawhoppen** [comment]
did you still not add an inaccuracy to the bullets

File: code/modules/projectiles/gun.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/gun.dm]: This should be on the gun itself, not every single gun.

File: code/modules/projectiles/gun.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/gun.dm]: implied `src`

File: code/modules/projectiles/gun.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/gun.dm]: There's a better way of handling things.

```
/obj/item/weapon/gun/gun_you_dont_want_to_update_icon/update_icon()
		return
```

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: Why is this even needed?

**Comment by Fox-McCloud** [comment]
Was this made in web editor or something. The fact it's displaying the amount of lines added is suspect.

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: `CB.forceMove(get_turf(src))`

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: no `\the`

File: code/modules/projectiles/guns/ballistic.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/ballistic.dm]: why do you need to exclude the ammo box from updating its icon? There's no unique ammo boxes here...likewise, even if there was, you could just click this on some other gun, and it would update.

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: If you're going to add an ammo box for this, you might as well add a crafting recipe for a makeshift 9mm ammo box.

**Comment by Fox-McCloud** [comment]
This uh, has a lot of coding problems with it, IMO.

**Comment by IlyaIU** [comment]
fox did you just reviewed pr from wrong repo or what

**Comment by Bawhoppen** [comment]
fox does stuff around here occasionally

**Comment by duncathan** [comment]
I hate pipes being used as weaponry TBH

**Comment by PKPenguin321** [comment]
Better remove the easier to make and objectively better pneumatic cannon then

In the meantime merge this

**Comment by duncathan** [comment]
don't tempt me to actually remove that pkp

**Comment by Fox-McCloud** [comment]
>I hate pipes being used as weaponry TBH

Does that mean you hate plasma-flooding, since the atmostechs are using the pipes as weapons? 🤔

**Comment by duncathan** [comment]
to be more specific I mean I hate `obj/item/pipe` being a component in weapons, not `obj/machinery/atmospherics/pipe` being used as a weapon by means of plasma flooding or my personal favorite, buckling to a heat exchange pipe

**Comment by lzimann** [comment]
No one seems to be interested in merging this, so I'll just close it.

**Comment by Bawhoppen** [comment]
we'll need to try this again under better circumstances and balance

**Comment by PKPenguin321** [comment]
Kor was thinking of merging it though

**Comment by duncathan** [comment]
He's more than welcome to re open it

**Comment by kevinz000** [comment]
what tacolizard gets to gunbloat with a resprited and nerfed stechtkin for greytiders and i don't get to add an actual good sniper rifle that's meant for long range use i'm salty now :(

**Comment by Repukan** [comment]
B O U N C I N G

A R R O W S

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (Tacolizard) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: it was needed for the variant sprites, but I never removed it.

File: code/modules/projectiles/guns/ballistic/revolver.dm
Comment (Tacolizard) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: it was copypasta

**Comment by Tacolizard** [comment]
ok so there's a ton of code issues and I probably won't be redoing this for a while.

**Comment by Tacolizard** [comment]
@Fox-McCloud I don't know what you mean by 'web editor'. I made this in DM. I do use Github Desktop if that would do anything.

**Pull request closed without merging**
